lib/goldfinger.rb

```diff
@@ -18,8 +18,10 @@ class NotFoundError < Error
   # @raise [Goldfinger::NotFoundError] Error raised when the Webfinger resource could not be retrieved
   # @raise [Goldfinger::SSLError] Error raised when there was a SSL error when fetching the resource
   # @param uri [String] A full resource identifier in the format acct:user@example.com
+  # @param opts [Hash] Options passed to HTTP.rb client
+  # @param ssl [Boolean] Whether performs secure HTTP reqest. MUST NOT set to false in the usual case
   # @return [Goldfinger::Result]
-  def self.finger(uri)
-    Goldfinger::Client.new(uri).finger
+  def self.finger(uri, opts = {}, ssl = true)
```

Comment:
I suggest dropping the `ssl` argument. Include it in `opts`, and then `ssl = opts.delete(:ssl)`